ClickHouse content

diawarr opened the pull request:
This is a draft pull request to allow previewing the content additions for Clickhouse.

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,40 @@
+
+ClickHouse supports SQL syntax that is mostly identical to the ANSI SQL standard. There are however some differences in behavior, which are listed in the `ClickHouse documentation <https://clickhouse.com/docs/en/sql-reference/ansi/>`_.
+
+The service is currently available as a beta release and is intended for **non-production use**.
```
How about we put it under a note or a warning?

diawarr replied:
True, that's now added as a note ;)

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,40 @@
+
+In OLAP scenarios, a column-oriented structure provides much quicker processing of queries compared to row-oriented databases, as it reduces the input/output load by minimizing the amount of data that needs to be read and by improving data compression. To optimize data compression, ClickHouse provides both general-purpose compression codecs and codecs that are tailored for specific data types. Cl ...
+
+ClickHouse supports SQL syntax that is mostly identical to the ANSI SQL standard. There are however some differences in behavior, which are listed in the `ClickHouse documentation <https://clickhouse.com/docs/en/sql-reference/ansi/>`_.
```
I have a feeling that "mostly identical" is a bit of an overstatement :D Maybe a word "resembles" would be better?

diawarr replied:
Yep, done ;)

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,40 @@
+
+* See the `main ClickHouse documentation <https://clickhouse.com/docs/en/>`_ for an introduction and a full list of features.
+
+* Our :doc:`getting-started` guide is a good way to get hands on with your first project..
```
```suggestion
* Our :doc:`getting-started` guide is a good way to get hands on with your first project.
```

lornajane replied:
This looks fixed now.

anelook commented on docs/products/clickhouse/getting-started.rst:
```diff
@@ -0,0 +1,45 @@
+
+      This defines the number of nodes, storage, memory, CPU, and backup support for the service.
+
+   d. Enter a name for your service.
```
The name cannot be changed later, should we specify this here? or a warning in the ui is enough?

diawarr replied:
I think it's enough that the UI includes a warning about this - I think we don't mention it in the documentation for the other services either.

anelook commented on docs/products/clickhouse/concepts/supported-table-engines.rst:
```diff
@@ -0,0 +1,63 @@
+
+  * - Engine
+    - Engine family
+  * - ``AggregatingMergeTree`` (remapped)
```
do we have somewhere info what "remapped" means? I can also clarify with with the ClickHouse team

diawarr replied:
Yep, that's now added as a note ;)

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,40 @@
+ClickHouse :badge:`beta,cls=badge-secondary text-black badge-pill`
+==================================================================
+
+What is Aiven for ClickHouse?
```
I believe we need to add "ClickHouse" to the list of the accepted terms

diawarr replied:
Actually, I think it should be OK - at least it's not reported as an issue in the checks...

diawarr replied:
That depends - at least it doesn't seem to raise any issues in the checks, but if we want to ensure the proper spelling, then it might be better to add "Clickhouse" to the rejected terms list?

anelook replied:
There is a warning which is raised :) It doesn't break and bring error, so I also almost missed it.

diawarr replied:
Ah, in that case it's better to add it - that's now in ;)

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,40 @@
+What is Aiven for ClickHouse?
+-----------------------------
+
+Aiven for ClickHouse beta is powered by ClickHouse, an open-source database management system that uses a column-oriented structure and is best suited for the online analytical processing of queries (OLAP).
```
one more thing which we can add is that, especially in case of clickhouse, which is a truly columnar database,  the values from the same column will be physically stored next to each other. That's why retrieving a column is so fast. On the other hand, the values belonging to the same raw are stored across different physical locations, so retrieving complete item rows is slow.

diawarr replied:
> Really enjoyed reading this!
> 
> I added couple of comments and suggestions. Also, I wonder if "Supported table engines" really belongs to concepts. But it doesn't feel into HowTos as well.

Thanks for the comments - the content is now updated based on those. And I agree, the concepts section might not be the best place for the "Supported table engines" article, so I've moved that under "References" ;)

anelook commented on .github/styles/Vocab/Docs/accept.txt:
```diff
@@ -8,6 +8,7 @@ Avro
 [Bb]oolean
 [Bb]oot
 [Bb]usiness
+[Cc]odecs
```
Are you sure we need this? I don't see we use "Codecs" with capital "C" anywhere

diawarr replied:
Not at the moment, but I just figured that it might be used for example at the beginning of a sentence at some stage, so the capitalized spelling might as well be allowed too ;)

anelook replied:
Ah, I think I see what you mean. Is "codecs" not a word from a normal vocabulary? You had to add "codecs", right? and [Cc] is just on top of it

diawarr replied:
Yep :)

anelook replied:
done

anelook reviewed (changes requested):
Really enjoyed reading this! 

I added couple of comments and suggestions.
Also, I wonder if "Supported table engines" really belongs to concepts. But it doesn't feel into HowTos as well.

diawarr commented:
Thanks for the comments @anelook - the content is now updated based on those. And I agree, the concepts section might not be the best place for the "Supported table engines" article, so I've moved that under "References" ;)

anelook commented:
@diawarr, thanks a lot! To me this reads very smoothly and up to the point. 
I'll create a draft for getting started with an example data set and then share with you for a review.

We also should add more about creating users with particular grants (we have it in console) and maybe about creating a fork (?). I can take care of it later, unless you come to it sooner :)

diawarr commented:
**Note:** This content should only be merged when the product is ready for release - current target is January 31st.

lornajane commented on docs/products/clickhouse/getting-started.rst:
```diff
@@ -0,0 +1,45 @@
+
+   The status is *Rebuilding* while the service is being created for you. Once the service is ready, the status changes to *Running*. While services typically start up in a couple of minutes, the performance varies between cloud providers and regions, and it may take longer in some circumstances.
+
+
```
I think we should revisit this page, creating a service does not get me started.

How about including an instruction to create a database, and showing an example of connecting with the docker version of the client command? Then leading people on immediately to the sample data, code samples if we have them, and other resources.

I don't see a reason to duplicate the "creating a service" page unless there's something different in the setup for this particular service that we want to mention. For example, linking to the generic create, and recommending some configuration or replication values seems more valuable to me.

diawarr replied:
Sure, that's now updated ;)

lornajane commented on docs/products/clickhouse/howto/add-service-users.rst:
```diff
@@ -0,0 +1,29 @@
+   .. note::
+      You cannot access this password later. However, you can reset the password for a user later if necessary.
+
+#. Close the summary view.
```
This seems a bit unnecessary, could we drop it?

lornajane commented on docs/products/clickhouse/howto/add-service-users.rst:
```diff
@@ -0,0 +1,29 @@
+
+   This creates the new user and shows you a summary of the information.
+
+#. Copy the password for the new user and paste it to a separate file.
```
I'm not sure we should be encouraging pasting into files. Let's combine this with the admonition below and just say something like "copy the password on screen to a safe place. It can't be accessed again in future, although it can be reset if needed" or similar.

anelook replied:
updated

lornajane commented on docs/products/clickhouse/index.rst:
```diff
@@ -0,0 +1,47 @@
+ClickHouse resources
+--------------------
+
+If you are new to ClickHouse, try these resources to get you started with the platform:
```
I feel sure we added this page first and there was nowhere to link to! Let's signpost the query browser as a good place to get started, and the sample datasets?

anelook replied:
added

lornajane reviewed (commented):
This is coming together nicely :) I've added a bunch of comments and suggestions, I think the focus this week should be the getting started page since that is likely to get a lot of attention when we launch. Please let me know if you need more help to flesh it out?

anelook commented on docs/products/clickhouse/sample-dataset.rst:
```diff
@@ -0,0 +1,153 @@
+
+When you create your service, a default database was already added. However, you can create separate databases specific to your use case. We will create a database with the name ``datasets``, keeping it the same as in the ClickHouse documentation.
+
+To create the new database, go to the  `Aiven web console <https://console.aiven.io/>`_ and click the **Databases & Tables** tab of your service page.
```
The database can be only created through the UI. The cli will not allow it. 

I can include more info on how to do it through ui - but the UI is very self explanatory.

lornajane replied:
Got it. I think that just a note to say "create the database `datasets`" would probably work here. We've made them click on the tab and then .... not prompted them to do the final step.

anelook replied:
updated

anelook commented on _toc.yml:
```diff
@@ -184,6 +184,26 @@ entries:
+      - file: docs/products/clickhouse/howto
+        title: HowTo
+        entries:
+          - file: docs/products/clickhouse/howto/add-service-users
```
done

lornajane commented:
I was writing such a lot of feedback on the main landing page that I gave up and edited it instead to try to guide our technical audience on what shape of problem ClickHouse is for. I'm not sure I've got the data analysis story strong here, but it should help the developer/implementor persona I think. Please check (especially for accuracy! We're all new to this tech) and amend as you see fit.

lornajane commented:
I also think we can consider a "work with databases and tables" HowTo article, so we can cover about creating tables through the Aiven interface (web/API) and _why_ that's the case. Bonus points if we capture the error message that users get if they try to do it in a normal ClickHouse way - this could be a common trip hazard for new-to-Aiven users.

anelook commented:
> I also think we can consider a "work with databases and tables" HowTo article, so we can cover about creating tables through the Aiven interface (web/API) and why that's the case. Bonus points if we capture the error message that users get if they try to do it in a normal ClickHouse way - this could be a common trip hazard for new-to-Aiven users.

@lornajane , could you clarify a bit more what you mean here? especially "capture the error message that users get if they try to do it in a normal ClickHouse way". 
Is there a specific way to create a table in Aiven Interface?

anelook commented:
But I like the idea of "work with databases and tables" it  can be a good one to mention that we ignore "on cluster" and the fact that you can only create database through interface.

lornajane commented:
Can we add the ClickHouse logo and links to the front page as well please?

anelook commented:
Done! I still need to verify that the svg image I created is okay from the point of view of designers. Or maybe they will give a better version.

anelook commented on docs/products/clickhouse/concepts/databases-and-tables.rst:
```diff
@@ -1,5 +1,5 @@
-Work with databases and tables
-===============================
+About databases and tables
```
I usually use format How To + verb. Here it is How to About databases. Do we care about How To + verb? or not that much?

lornajane replied:
I moved it to concepts, because it wasn't very task-based. So then the titles often are "About ...."

anelook commented on docs/products/clickhouse/concepts/databases-and-tables.rst:
```diff
@@ -10,23 +10,21 @@ Databases and tables are at the core of any **Database Management System**. Clic
 ---------
 
-Similarly when working with any other database technology, the first step is to create the database itself. Due to security reasoning, the `admin` user is not allowed to create databases. However, you can create a new database through the web interface of `Aiven console <https://console.aiven.io/>`_:
+Creating tables in an Aiven for ClickHouse service can only be done via the Aiven platform; the `admin` user is not allowed to create databases directly for security and reliability reasons. However, you can create a new database through the web interface of `Aiven console <https://console.aiven.io/>`_:
```
This is a bit confusing - "Creating tables" - we're talking about databases here
so it probably should be "Creating databases in an Aiven for ClickHouse service can only be done via "
To make it more specific I'd replace platform with console.

lornajane replied:
Good catch!

anelook commented on docs/products/clickhouse/concepts/databases-and-tables.rst:
```diff
@@ -10,23 +10,21 @@ Databases and tables are at the core of any **Database Management System**. Clic
 .. note::
 
-    If you try adding a new database to **Aiven for ClickHouse** through the command line, you'll encounter an exception ``Not enough privileges.(ACCESS_DENIED)``. Please use the web interface to add a new database.
+    If you try adding a new database to **Aiven for ClickHouse** through the command line, you'll encounter an exception ``Not enough privileges.(ACCESS_DENIED)``. Please use the Aiven platform, either the web interface or one of our other tools, to add a new database.
```
" or one of our other tools" - what are the other tools?

lornajane replied:
THERE it is, I knew I'd edited this in another place and I couldn't locate it with keywords like "API". Thanks!

anelook commented on docs/products/clickhouse/index.rst:
```diff
@@ -54,4 +54,4 @@ If you are new to ClickHouse, try these resources to get you started with the pl
 * Follow :doc:`a guide to add a sample dataset <sample-dataset>` to your service.
 
-* Find out :doc:`how to use the web-based query editor <howto/use-query-editor>`, that comes as part of ClickHouse service.
+* Find out :doc:`how to use the web-based query editor <howto/use-query-editor>`, that comes as part of the Aiven for ClickHouse service.
```
Technically it is part of  ClickHouse service. We didn't develop it, so I though "of ClickHouse service" is more appropriate

lornajane replied:
I made it "of the ClickHouse service", thanks for the clarification
